collector/adcs.go

```diff
@@ -0,0 +1,246 @@
+//go:build windows
+// +build windows
+
+package collector
+
+import (
+	"errors"
+	"github.com/prometheus-community/windows_exporter/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"strings"
+)
+
+func init() {
+	registerCollector("adcs", ADCSCollectorMethod, "Certification Authority")
+}
+
+type ADCSCollector struct {
+	RequestsPerSecond                            *prometheus.Desc
+	RequestProcessingTime                        *prometheus.Desc
+	RetrievalsPerSecond                          *prometheus.Desc
+	RetrievalProcessingTime                      *prometheus.Desc
+	FailedRequestsPerSecond                      *prometheus.Desc
+	IssuedRequestsPerSecond                      *prometheus.Desc
+	PendingRequestsPerSecond                     *prometheus.Desc
+	RequestCryptographicSigningTime              *prometheus.Desc
+	RequestPolicyModuleProcessingTime            *prometheus.Desc
+	ChallengeResponsesPerSecond                  *prometheus.Desc
+	ChallengeResponseProcessingTime              *prometheus.Desc
+	SignedCertificateTimestampListsPerSecond     *prometheus.Desc
+	SignedCertificateTimestampListProcessingTime *prometheus.Desc
+}
+
+// ADCSCollectorMethod ...
+func ADCSCollectorMethod() (Collector, error) {
+	const subsystem = "adcs"
+	return &ADCSCollector{
+		RequestsPerSecond: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "requests_total"),
+			"Certificate requests processed per second",
+			[]string{"cert_template"},
+			nil,
```

Comment:
`requests_total` indicates this is the total number of requests, but the description states this metric is a per-second rate.

There's a number of other metrics with a similar issue, but one comment thread is adequate for this.

Reply:
Metric description updated due to Windows reporting the *last* known value, not an average, etc.